vizro-core/examples/_dev/app.py

```diff
@@ -45,7 +47,95 @@
     ],
 )
 
-dashboard = vm.Dashboard(pages=[cards, graph], navigation=vm.Navigation(nav_selector=vm.NavBar()))
+# CREATE FAKE DATA
+column = [1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4]
+row = [1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5]
+value = ["1", "7", "3", "5", "2", "10", "9", "1", "3", "7", "5", "8", "2", "9", "1", "2", "7", "5", "3", "2"]
+group = ["A", "B", "C", "D", "E", "E", "D", "C", "B", "A", "A", "E", "C", "B", "D", "A", "D", "B", "C", "E"]
+date_time = [
+    "2023-01-01",
+    "2023-02-02",
+    "2023-03-03",
+    "2023-04-04",
+    "2023-05-05",
+    "2023-06-06",
+    "2023-07-07",
+    "2023-08-08",
+    "2023-09-09",
+    "2023-10-10",
+    "2023-11-11",
+    "2023-12-12",
+    "2024-01-01",
+    "2024-02-02",
+    "2024-03-03",
+    "2024-04-04",
+    "2024-05-05",
+    "2024-06-06",
+    "2024-07-07",
+    "2024-08-08",
+]
+
+data = pd.DataFrame()
+
+# date_time_new = [datetime.strptime(date, "%Y-%m-%d") for date in date_time]
+
+data["COLUMN1"] = column
+data["ROW1"] = row
+data["VALUE"] = value
+data["GROUP"] = group
+data["DATE_TIME"] = date_time
+
+# data["DATE_TIME"] = pd.to_datetime(data["DATE_TIME"])
+
+page_1 = vm.Page(
+    title="Datepicker page",
+    components=[vm.Table(id="table", figure=dash_data_table(data_frame=data))],
+    controls=[
+        vm.Filter(
+            column="DATE_TIME",
+            selector=vm.DatePicker(
+                title="Pick a date",
+                min="2023-01-01",
+                value=["2024-01-01", "2024-03-01"],
+                max="2024-07-07",
+            ),
+            # selector=vm.DatePicker(title="Pick a date", min_date='2023-01-01', value=['2024-01-01'], multi=False),
+        )
+    ],
+)
```

Comment:
I tried out numerous combinations and it works so smoothly 🚀. 

The only inconsistency I found is when I configure these two cases:
- multi=False, value=["2024-01-01", "2024-03-01"] and
- multi=True, value="2024-01-01"

So we should probably validate it.

Reply:
I just found another weird behaviour. Both date pickers behave strange when the value is cleared, and when you navigate to another page and then get back to the page where the date picker is cleared.
- single date picker returns a default value (instead of being cleared as the vm.Dropdown does)
- rande date picker raises an exception

Reply:
Unfortunately, the bug you found is on the dmc component, as it produces the exact same exception when using pure dash. 
Possible option to mitigate this, is set `clearable=False` and disable clearing the datepicker. Pro's of this approach is we keep using dmc component and keep `persistence=True`.

Reply:
That's fine for me 👍 But I remember that when I proposed that during stand-up, that there would be still a second issue remaining? Is this not the case anymore?

If we just need to set clearable=False, then I'm happy to do that 👍 Just add a comment of why we have it there and raise an issue in the dmc library as well. So in case they do fix it, we can also remove it :)

Reply:
I agree with the workaround you suggested (`clearable=False` and `persistence=True`). 

However, if there is no rush to release the DatePicker feature, I would suggest that we try to solve this problem in a similar way to the one that appears in `vm.Slider` and `vm.RangeSlider`: 
Attempting to implement a client-side callback that is triggered on a `date_picker.value` change, with inputs=[`date_picker.value`, `temp_store.data`] with outputs=[`date_picker.value`] and a body that checks if the `date_picker_value` is null, and if it is, assign the `temp_store_data` to its value. 

I'm not a wizard in the dmc bug you found but I give 50% chance that this approach can solve the problem we have.

Reply:
There is no rush, so happy for you guys to try that out if you think it could work 👍